src/instructlab/utils.py

```diff
@@ -564,3 +596,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
```

Comment:
Could you please add a link to where the HF format is defined, for reference?

Reply:
Sure, I updated it 🙂.